
The Apple USB keyboard does send non-standard events that do not make working
that comfortable. That is we swap the events here as in the 95-adb-mapping
patch.

  - Rene Rebe <rene@exactcode.de>

--- linux-2.6.10/drivers/usb/input/hid-input.c	2004-12-24 22:34:32.000000000 +0100
+++ linux-2.6.10-fixed/drivers/usb/input/hid-input.c	2005-02-25 13:14:35.000000000 +0100
@@ -107,6 +107,14 @@
 			if ((usage->hid & HID_USAGE) < 256) {
 				if (!(usage->code = hid_keyboard[usage->hid & HID_USAGE]))
 					return;
+				if (hidinput->input.id.vendor == 0x5ac &&
+				    hidinput->input.id.product == 0x020c) {
+					switch (usage->code) {
+					  case 56: usage->code = 100; break;
+					  case 125: usage->code = 56; break;
+					  case 126: usage->code = 56; break;
+					}
+				}
 				clear_bit(usage->code, bit);
 			} else
 				usage->code = KEY_UNKNOWN;
